index.js

```diff
@@ -1,18 +1,64 @@
+// Imports
+const axios = require("axios");
+const cors = require("cors");
 const express = require("express");
-const appUserRoute = require("./routes/appUsersRoutes");
-const appPostRoute = require("./routes/postRoutes");
+const { expressjwt: jwt } = require("express-jwt");
+const jwks = require("jwks-rsa");
+
+const pool = require("./database/db");
+const userRoute = require("./routes/userRoutes");
+const postRoute = require("./routes/postRoutes");
+
+const verifyJwt = jwt({
+  secret: jwks.expressJwtSecret({
+    cache: true,
+    rateLimit: true,
+    jwksRequestsPerMinute: 5,
+    jwksUri: `https://${process.env.AUTH0_DOMAIN}/.well-known/jwks.json`,
+  }),
+  audience: process.env.AUTH0_AUDIENCE,
+  issuer: `https://${process.env.AUTH0_DOMAIN}/`,
+  algorithms: ["RS256"],
+}).unless({ path: ["/"] });
+
+const currentUserCheck = async (req, res, next) => {
+  const auth0Id = req.auth.sub;
+  const query = await pool.query(
+    `SELECT * FROM app_user WHERE "auth0Id" = $1`,
+    [auth0Id]
+  );
+  const user = query.rows;
+  if (user.length == 0) {
+    try {
+      const accessToken = req.headers.authorization.split(" ")[1];
+      const response = await axios.get(
+        "https://dev-gxkwzphy3vwb5jqh.us.auth0.com/userinfo",
```

Comment:
Is this tenant's name publicly available? I was wondering if maybe we should make this another .env variable if its something we should not be showing publicly

Reply:
As discussed offline, I see your point, but since this is just our project for fun and given that this is also already in the git history, we can just leave it. You need an access token to use this endpoint anyways, so it's probably not that big of an issue